sdk/azidentity/aad_identity_client.go

```diff
@@ -141,7 +142,11 @@ func (c *aadIdentityClient) createAccessToken(res *azcore.Response) (*azcore.Acc
 		ExpiresIn json.Number `json:"expires_in"`
 		ExpiresOn string      `json:"expires_on"`
 	}{}
-	if err := json.Unmarshal(res.Payload, &value); err != nil {
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(body, &value); err != nil {
```

Comment:
You can replace `json.Unmarshal()` with `res.UnmarshalAsJSON()`.  You won't have to call `ioutil.ReadAll()` thus avoiding an extra copy of the underlying data.